doc/source/train/user-guides/experiment-tracking.rst

```diff
@@ -1,195 +1,326 @@
-.. _train-experiment-tracking:
+.. _train-experiment-tracking-native:
 
+===================
 Experiment Tracking
 ===================
 
-Most experiment tracking libraries work out-of-the box with Ray Train. This means
-you can use the regular tracking library APIs to report your results, models, and
-other artifacts.
+.. note::
+    This guide is relevant for all DataParallelTrainer, including TorchTrainer and TensorflowTrainer.
 
-Ray Train also exposes logging callbacks that automate some of these tasks.
+Most experiment tracking libraries work out-of-the-box with Ray Train. 
+This guide provides instructions on how to set up the code so that your favorite experiment tracking libraries 
+can work for distributed data parallel training with Ray Train. We will conclude the session with debugging
+tips.
 
-.. _train-experiment-tracking-native:
+Before we begin, the following is roughly how you can use the native experiment tracking lirary calls 
+inside of Ray Train. 
 
-Using native experiment tracking libraries
-------------------------------------------
+.. code-block:: python
 
-.. tab-set::
+    from ray.train.torch import TorchTrainer
+    from ray.train import ScalingConfig
 
-    .. tab-item:: PyTorch
+    def train_func(config):
+        # Training code and native lexperiment tracking library calls go here.
 
-        You can use experiment tracking libraries such as Weights & Biases, Mlflow, or
-        Comet directly in your Ray Train training loop.
+    scaling_config = ScalingConfig(num_workers=2, use_gpu=True)
+    trainer = TorchTrainer(train_func, scaling_config=scaling_config)
+    result = trainer.fit()
 
-        There are two things to keep in mind:
+Ray Train lets you use native experiment tracking libraries by customizing the tracking 
+logic inside the ``train_func`` function. In this way, you can port your experiment tracking 
+logic to Ray Train with minimal changes. 
 
-        1. Your code is executed in parallel on many workers. However, you often only want to report
-        results from one of these workers (usually the first worker - the "rank 0" worker).
+Log distributed training experiments
+====================================
 
-        2. When using the native libraries, you should report the results to both Ray Train and
-        the experiment tracking library.
+For some scenarios, every Worker generates an identical copy and saving a single copy is sufficient. 
+Ray Train lets you apply logic to only the rank 0 Worker with the following method:
+:meth:`context.get_world_rank() <ray.train.context.TrainContext.get_world_rank>`.
 
-        Example:
+There are roughly two kinds of experiment tracking modes: online and offline. For online mode, you log
+towards a tracking service that is running. Usually you need credentials to access the service.
+For offline mode, you log towards local file directory. Usually no credentials are needed.
```

Comment:
I don't think this is very central to experiment tracking. I believe most users are using online logging, and we should optimize the guide for those.

It's still worth pointing out that all nodes should have access to the credentials

Reply:
I reworked this section quite a bit. it still touches upon offline a bit (just because I find some users do use this for development) but not too much.